src/detectors/builtin/sendInLoop.ts

```diff
@@ -0,0 +1,297 @@
+import { CompilationUnit } from "../../internals/ir";
+import { foldStatements } from "../../internals/tact";
+import { MistiTactWarning, Severity } from "../../internals/warnings";
+import { ASTDetector } from "../detector";
+import {
+  AstStatement,
+  AstStatementExpression,
+  AstExpression,
+  AstCondition,
+  AstStatementTry,
+  AstStatementTryCatch,
+  AstStatementWhile,
+  AstStatementRepeat,
+  AstStatementUntil,
+  AstStatementForEach,
+  AstNumber,
+} from "@tact-lang/compiler/dist/grammar/ast";
+
+/**
+ * A detector that identifies send function calls inside loops and loops that may cause out-of-gas exceptions.
+ *
+ * ## Why is it bad?
+ * **Send Functions Inside Loops:**
+ * - **Unexpected Behavior:** Multiple sends may occur, leading to unintended contract interactions.
+ * - **High Gas Consumption:** Loops with send calls can consume excessive gas, potentially causing out-of-gas exceptions.
+ * - **Reentrancy Vulnerabilities:** Sending messages within loops might open up reentrancy attack vectors.
```

Comment:
Do we have reentrancy in TON?

Reply:
Yes we don't have yet this. Removed reentrancy part.